pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -0,0 +1,431 @@
+    });
+
+    sharedBeforeEach('set delay for action', async () => {
+      await authorizer.scheduleAndExecuteDelayChange(action, delay, { from: root });
+    });
+
+    const schedule = async (
+      where: Contract,
+      executors: SignerWithAddress[] | undefined = undefined
+    ): Promise<number> => {
+      data = authenticatedContract.interface.encodeFunctionData('protectedFunction', [functionData]);
+      return authorizer.schedule(where, data, executors || [], { from: user });
+    };
+
+    it('schedules a non-protected execution', async () => {
+      const id = await schedule(authenticatedContract);
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.executed).to.be.false;
+      expect(scheduledExecution.data).to.be.equal(data);
+      expect(scheduledExecution.where).to.be.equal(authenticatedContract.address);
+      expect(scheduledExecution.protected).to.be.false;
+      expect(scheduledExecution.executableAt).to.be.at.almostEqual((await currentTimestamp()).add(delay));
+    });
+
+    it('can schedule with a global permission', async () => {
+      await authorizer.revokePermission(action, user, authenticatedContract, { from: root });
+      await authorizer.grantPermission(action, user, EVERYWHERE, { from: root });
+      const id = await schedule(authenticatedContract);
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.executed).to.be.false;
+      expect(scheduledExecution.data).to.be.equal(data);
+      expect(scheduledExecution.where).to.be.equal(authenticatedContract.address);
+      expect(scheduledExecution.protected).to.be.false;
+      expect(scheduledExecution.executableAt).to.be.at.almostEqual((await currentTimestamp()).add(delay));
+    });
+
+    it('cannot execute the action immediately', async () => {
+      const id = await schedule(authenticatedContract);
+      await expect(authorizer.execute(id)).to.be.revertedWith('ACTION_NOT_YET_EXECUTABLE');
+    });
+
+    it('can be executed by anyone', async () => {
+      const id = await schedule(authenticatedContract);
+      await advanceTime(delay);
+
+      const receipt = await authorizer.execute(id);
+      expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', { scheduledExecutionId: id });
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.executed).to.be.true;
+
+      expectEvent.inIndirectReceipt(await receipt.wait(), authenticatedContract.interface, 'ProtectedFunctionCalled', {
+        data: functionData,
+      });
+    });
+
+    it('cannot be executed twice', async () => {
+      const id = await schedule(authenticatedContract);
+      await advanceTime(delay);
+
+      await authorizer.execute(id);
+      await expect(authorizer.execute(id)).to.be.revertedWith('ACTION_ALREADY_EXECUTED');
+    });
```

Comment:
They certainly are but we tend to have overlap in tests suites of close functions, for example, schedule/execute, scheduleGrantPermission/grantPermission. I am neutral on this. If you want to remove this let me know.

Reply:
I'd remove these because a) we have the same exact test for `execute`, so they don't really add anything, and b) reading these caused me some confusion as I started wondering what we should test in each `describe` block, thinking about other missing `execute` tests, going to that section and double-checking coverage, etc., which made it harder for me to be confident about correctness of the test suite.